pulp/apis/cuopt_api.py

```diff
@@ -0,0 +1,337 @@
+import ctypes
+import os
+import subprocess
+import sys
+import warnings
+from uuid import uuid4
+import numpy as np
+from ..constants import (
+    LpBinary,
+    LpConstraintEQ,
+    LpConstraintGE,
+    LpConstraintLE,
+    LpContinuous,
+    LpInteger,
+    LpMaximize,
+    LpMinimize,
+    LpStatusInfeasible,
+    LpStatusNotSolved,
+    LpStatusOptimal,
+    LpStatusUnbounded,
+    LpStatusUndefined,
+)
+from .core import (
+    LpSolver,
+    LpSolver_CMD,
+    PulpSolverError,
+    clock,
+    ctypesArrayFill,
+    sparse,
+)
+
+# COPT string convention
+if sys.version_info >= (3, 0):
+    coptstr = lambda x: bytes(x, "utf-8")
+else:
+    coptstr = lambda x: x
+
+byref = ctypes.byref
+
+# Constraint Sense Converter
+sense_conv = {LpConstraintLE: "L",
+              LpConstraintGE: "G",
+              LpConstraintEQ: "E",
+             }
+
+def process_constraint(args):
+    local_data = []
+    local_vars = []
+    local_sense = []
+    local_rhs = []
+    for constraints in args:
+        name, constraint = constraints
+        local_data.extend(list(constraint.values()))
+        local_vars.extend(list(constraint.keys()))
```

Comment:
I think the order of keys and values dictionaries in python is preserved so this code *should* run fine.
Having said that, I think I prefer the following code, that takes values and keys in one go, in the remote case the order is not preserved for whatever reason.
```python
data, vars = zip(*a.items())
local_data.extend(data)
local_vars.extend(vars)
```

Reply:
Removed in #839